src/tools/ScenarioMeasurement/Startup/PerfCollect.cs

```diff
@@ -147,7 +148,14 @@ public void AddKernelKeyword(KernelKeyword keyword)
 
         private bool LttngInstalled()
         {
-            return File.Exists("//usr/bin/lttng");
+            ProcessStartInfo procStartInfo = new ProcessStartInfo("modinfo", "lttng_probe_writeback");
+            Process proc = new Process() { StartInfo = procStartInfo, };
+            proc.StartInfo.RedirectStandardOutput = true;
+            proc.Start();
+            proc.WaitForExit();
+            string result = proc.StandardOutput.ReadToEnd();
```

Comment:
This pattern is prone to deadlocking if the process writes enough output to the buffer. Consider using a pattern as we do in `ProcessHelper`:
https://github.com/dotnet/performance/blob/90b537906d686846ec7c8293cefd268e0143452c/src/tools/ScenarioMeasurement/Util/ProcessHelper.cs#L222-L243

if you're confident the output of `modinfo` will not get very long this is fine. It's something to be aware of.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.standardoutput?view=net-5.0
seems to indicate that moving string result = proc.StandardOutput.ReadToEnd(); before proc.WaitForExit(); is sufficient to avoid the deadlock problem.  I've tested it and it seems to run fine both for code tests and in startup tests.

Reply:
Since the volume of output returned by modinfo is pretty consistent, I don't think any further measures are necessary.